### ISSUE TITLE
adding koreader to home tiles?

### ISSUE THREAD

**Issue by anarcat** [koreader/koreader#1976]
I went through the install process documented in the wiki here to setup Koreader on my Kobo Glo HD. I have KSM working on reboot, and it seems I am able to start koreader from there.

Maybe I completely misunderstand how things work here, but I was expecting koreader to be available from the main Kobo interface (which I understand is called "nickel"?). I am not sure I want to always use koreader to read books, it's more an auxiliary tool for me right now... This is particularly because the rendering for the book I am reading right now is not as good as the one from the default reader in Kobo (for example, headings are not formatted properly).

Is it possible to access koreader without rebooting my Kobo? Sorry if this is a FAQ, but I couldn't find it in the wiki.

**Comment by KenMaltby** [comment]
Yes, there are KSM options that can enable a Nickel to KOReader swapping.  See the KSM8 thread at MobileRead forums: http://www.mobileread.com/forums/showthread.php?t=266821 
If, as you say, you seldom use KOReader you might want to set KSM to auto launch in Nickel.

**Comment by anarcat** [comment]
hmm... there are 10 pages of comments in there. i already searched for "tile" and "home" and couldn't find the answer I was looking for.

**Comment by KenMaltby** [comment]
Try poking "configure" on the main KSM menu, then poke [koreader] > then poke "insertNickelKoreaderSwitch=false" > then poke "insertNickelKoreaderSwitch=true"

**Comment by anarcat** [comment]
Okay, so I did that and i actually had to do "addItem" to add `insertNickelKoreaderSwitch=true`, as it was not present already. i did save and return and reboot, then started nickel. i don't see anything changed on the home screen. 

maybe i am not looking at the right place?

and what will this do?

to be clear, what I am looking for here is a way to start koreader directly from the Nickel home screen, without going through KSM.

if there's no way to do that, a way to reboot the machine more easily would be good: right now I need to hold the power button for like 10 seconds twice to reboot...

**Comment by anarcat** [comment]
also, I followed http://www.mobileread.com/forums/showpost.php?p=3112371&postcount=301 to the letter, and the menus don't show up in KSM8.

**Comment by KenMaltby** [comment]
If you are insisting on using the "fmon" approach to launch from within Nickel, This Guide should help: http://www.mobileread.com/forums/showthread.php?t=226456 

 The insertNickelKoreaderSwitch=true should give you an option in the KSM menu to start in Nickel and when you leave Nickel you go to KOReader.  Also, when you then leave KOReader you return to Nickel.
You can also add it as the autostart, to automatically start in that mode.   

"also, I followed http://www.mobileread.com/forums/showpost.php?p=3112371&postcount=301 to the letter, and the menus don't show up in KSM8."
This is a way to add back the "extra features" that previous versions of the Kobo firmware had.  They should show up in Nickel, not KSM.

**Comment by anarcat** [comment]
okay well, i don't know what "fmon" is. i do not mean to insist on anything, but i do want to keep running Nickel for now, and it'd be great if i could start stuff directly from there, so i guess the "fmon" approach is the way to go for me, even though I don't understand what it means - is there a glossary of that stuff somewhere?

anyways - i went through the guide in the forum, and I got the launcher.png image to display, but at first the launcher wouldn't work. it seems i am *also* supposed to copy the `vlasovsoft` folder, although the instructions just say to transfer only the `KoboRoot.tgz` file. if the `vlasovsoft` folder is not present, the launcher doesn't work and the launcher icon is just an image. the background of the launcher seems to turn to black when it succeeds, is that possible?

it is also unclear at 12 which `KoboRoot.tgz` needs to be copied - is it the one from koreader or pbchess? i tried the one from koreader and, after step 12 the machine rebooted and the tiles on the home screen were all weird. tapping on one of them would go to a weird library purchase dialog where i could seemingly purchase an unknown book. i removed those tiles from the welcome screen to clear things up (that was a mistake!). it looks like the books i was reading were corrupted somehow. and of course the launchers were broken at that point. removing and re-adding the launcher.png icon fixed the pbchess shortcut. tapping the launcher then works, but it disappears again on the next reboot. it turns out that removing the tiles made those things disappear completely and definitely.

Something really fishy is going on here, i am not sure i understand what i am doing... 

it would be great if those instructions would be in this wiki instead of spread all around the forums... it's pretty hard to find what you are looking for as a new user, and there's a whole genealogy of terms to get familiar with as well...

**Comment by anarcat** [comment]
oh, and should i be posting this in the forums instead? i felt this was a documentation issue here, but i can also go back to the forum if you prefer that. the idea here was to build more stable documentation here in the wiki or the README file...

**Comment by KenMaltby** [comment]
You are reporting some very odd behavior, none of which should be the result of following the guide.
Assuming that your KSM8 is installed properly, you should be able to launch into Nickel and you should be able to launch into KOReader.  You should also have tiles/books/pic in Nickel's library, that launch the functions in their file names:
exit_nickel.png
poweroff.png
run_test_script.png
simulate_usb_connection.png
switchtokoreader.png
toggle_nightmode.png

exit_nickel should return you to KSM8, without a reboot, and you can then select koreader.  Or if you started Nickel using the Nickel_KOReader_Switch in the KSM menu, you can just use the "switchtokoreader.png" icon.

BUT- it sounds to me like you should remove everything and start over.

**Comment by anarcat** [comment]
I believe [the guide](http://www.mobileread.com/forums/showthread.php?t=226456) is incomplete. It doesn't say that the vlasovsoft directory should be copied, and doesn't clearly say which .tgz should be copied on step 12. I am not sure how others managed to follow those instructions, I can only have admiration for those people. :)

Honestly at this point, i have just given up. I was looking for OPDS support, but the other side of things (https://github.com/wallabag/wallabag/issues/1253) doesn't have OPDS support yet so i'll just postpone this struggle to another time.

**Comment by tigran123** [comment]
@anarcat I didn't follow any instructions, but I use (and have been using for many years) KSM with Koreader and CoolReader (pbchess package from vlasovsoft) without any problems whatsoever. The key thing to remember is to use only KSM 06 (older versions are broken, newer versions are not needed, because when something is working perfectly one should stop the development and actually use the software). As for Koreader and pbchess --- I am using the latest versions of both (2015-stable Koreader and 2016.03 of pbchess) and they are working perfectly. This is on all three of my Kobo devices: Kobo Mini, Kobo Aura H2O and Kobo Aura HD.

Basically, the instruction is this:

1. Install pbchess (but don't try to execute it just yet) in Nickel.

2. Install KSM 06.

3. Copy koreader directory to .kobo (no need to "install" anything as Koreader doesn't change anything in root filesystem).

4. Hack `.kobo/kbmenu/onstart/start_vlasovlauncher.sh` to NOT exec `vlasovsoft/launcher.sh` (because it contains a lot of unnecessary stuff that wastes valuable time or increases data corruption risk, e.g. remounting microsd for rw, unneeded reboot on exit etc etc) Instead, make it short like this:

#!/bin/sh
export ROOT=/mnt/onboard/.kobo/vlasovsoft
LOG=/dev/null
export DEVICE=AURAH2O
export TMPDIR=/tmp/vlasovsoft
export LANG=en_US.UTF-8
export VLASOVSOFT_KEY=$ROOT/key
export VLASOVSOFT_KBD=$ROOT/kbd.txt
export VLASOVSOFT_FIFO1=$TMPDIR/fifo1
export VLASOVSOFT_FIFO2=$TMPDIR/fifo2
export VLASOVSOFT_I18N=$ROOT/i18n
export VLASOVSOFT_DICT=$ROOT/dictionary
export QT_PLUGIN_PATH=$ROOT/Qt/plugins
export LD_LIBRARY_PATH=$ROOT/Qt/lib
export QWS_MOUSE_PROTO=KoboTS_h2o
export QWS_KEYBOARD=KoboKb
export QWS_DISPLAY=Transformed:KoboFB
export QT_QWS_FONTDIR=$ROOT/fonts
export STYLESHEET=$ROOT/eink.qss
export UPDATE_DELAY_PART=25
export UPDATE_DELAY_FULL=50
export SWIPE_MIN_DISTANCE=12 # mm
cd $ROOT
mkdir -p $TMPDIR
mkfifo $VLASOVSOFT_FIFO1
mkfifo $VLASOVSOFT_FIFO2
$ROOT/launcher -qws -stylesheet $STYLESHEET > $LOG 2>&1
rm $VLASOVSOFT_FIFO1
rm $VLASOVSOFT_FIFO2

This is for H2O (for other devices edit it accordingly, look inside `vlasovsoft/launcher.sh` to understand what values of `DEVICE` and other variables to change).

Also, you will need to edit `eink.qss` and `cr3/data/fb2.css` and `cr3/data/epub.css` as the defaults there are not useable (buttons too big etc etc).

**Comment by nshram** [comment]
An old thread.. and maybe the original poster has found the solution..
I'm a new Kobo user (Aura One!) and I had the same issue.
I'd installed KSM v8 plus the patch for KA1.
After this, I just enabled the koreader to nickel and nickel to koreader switches from the koreader additions sub-menu in KSM home.
Now there a quick and seamless way to switch between both!
In nickel, the "switchtokoreader" image has to be selected..
And in koreader, selecting exit brings back nickel, instead of KSM.

**Comment by Frenzie** [comment]
Yes, that's how it's supposed to work. It's still unclear to me what went wrong in the OP.

**Comment by nshram** [comment]
Well... the setting doesn't seem to be persistent across reboots.
I now find that switching from Nickel or Koreader lands me in the KSM home screen. From here, I have to select the koreader or nickel options.
I actually wanted this option too, separately!
The original option returns after again poking the Koreader sub menu options.
Not a big hassle.. but I don't like selecting an icon which says "Koreader"(sideways!), when it's actually going to send me to the KSM home!

**Comment by Frenzie** [comment]
I'm not sure what you mean by setting? After booting into KSM if I select `KOReader additions` → `Start … switch` it always works just fine.

If you mean you want to auto-start some KSM menu entry that's a different matter. You can set an auto select option somewhere but I guess you mean the switch isn't available for that?

**Comment by KenMaltby** [comment]
nshram; you might want to address KSM issues to the KSM author at https://www.mobileread.com/forums/showthread.php?t=266821
As I remember those setting they were persistent and remained in effect through a reboot.
